File: src/main/java/com/javatechie/webflux/handler/CustomerHandler.java
package com.javatechie.webflux.handler;

import com.javatechie.webflux.dao.CustomerDao;
import com.javatechie.webflux.dto.Customer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.web.reactive.function.server.ServerRequest;
import org.springframework.web.reactive.function.server.ServerResponse;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Service
public class CustomerHandler {

    @Autowired
    private CustomerDao dao;


    public Mono<ServerResponse> loadCustomers(ServerRequest request){
        Flux<Customer> customerList = dao.getCustomerList();
        return ServerResponse.ok().body(customerList,Customer.class);
    }


    public Mono<ServerResponse> findCustomer(ServerRequest request){
      int customerId= Integer.valueOf( request.pathVariable("input"));
       // dao.getCustomerList().filter(c->c.getId()==customerId).take(1).single();
        Mono<Customer> customerMono = dao.getCustomerList().filter(c -> c.getId() == customerId).next();
        return ServerResponse.ok().body(customerMono,Customer.class);
    }


    public Mono<ServerResponse> saveCustomer(ServerRequest request){
        Mono<Customer> customerMono = request.bodyToMono(Customer.class);
        Mono<String> saveResponse = customerMono.map(dto -> dto.getId() + ":" + dto.getName());
        return ServerResponse.ok().body(saveResponse,String.class);
    }



}

File: springboot-webflux-demo-Kotlin/src/main/kotlin/com/myproject/webflux/handler/CustomerHandler.kt
package com.myproject.webflux.handler

import org.springframework.web.reactive.function.server.ServerResponse

import reactor.core.publisher.Mono

import com.myproject.webflux.dto.Customer

import org.springframework.web.reactive.function.server.ServerRequest

import reactor.core.publisher.Flux

import com.myproject.webflux.dao.CustomerDao

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.util.function.Function


@Service
class CustomerHandler {
    @Autowired
    private val dao: CustomerDao? = null
    fun loadCustomers(request: ServerRequest?): Mono<ServerResponse> {
        val customerList: Flux<Customer> = dao!!.getCustomerList()
        return ServerResponse.ok().body(customerList, Customer::class.java)
    }

    fun findCustomer(request: ServerRequest): Mono<ServerResponse> {
        val customerId = Integer.valueOf(request.pathVariable("input"))
        // dao.getCustomerList().filter(c->c.getId()==customerId).take(1).single();
        val customerMono: Mono<Customer> = dao!!.getCustomerList().filter { c->c.id == customerId }.next()
        return ServerResponse.ok().body(customerMono, Customer::class.java)
    }

    fun saveCustomer(request: ServerRequest): Mono<ServerResponse> {
        val customerMono = request.bodyToMono(Customer::class.java)
        val saveResponse = customerMono.map { dto: Customer -> dto.id.toString() + ":" + dto.name  }
        return ServerResponse.ok().body(saveResponse, String::class.java)
    }
}
